### DISCUSSION TITLE
Try to clean ELF header from generated executable when compiling for Android

### DISCUSSION THREAD

**Discussion by fabricio-p** [vlang/v#11912] [Ideas]
When the C compiler generates an executable on/for Android, the executable includes what is called an ELF header, and everytime you run it, it will always prints this
```
WARNING: linker: /some/path/to/executable: unsupported flags DT_FLAGS_1=0x8000001
```
When compiling C, you can clean the ELF header using the Makefile, but V has its own build system and strategy.
If the compilation is happening inside an Android device and the `termux-elf-cleaner` command is detected, run that with the path of the executable as the first argument. You can detect if the device is Android by check if one of the environment variables `ANDROID_DATA` or `ANDROID_PATH` are set.

**Comment by larpon** [comment]
@fabriciopashaj - can you provide a minimal project or some source code that I can run in `termux` to reproduce these messages?
(also the commands to get a working V compiler in termux would help me a lot!).
I don't use termux personally so some guide on how to kickstart V development would be highly appreciated (maybe we can even include it in the official docs if made short and general enough, not sure)

**Comment by fabricio-p** [comment]
Of course.
Run those commands to setup
```bash
apt update && apt upgrade
apt install git clang termux-elf-cleaner
```
If you create a, i.e. C file named main.c with the contents below
```c
int main() {
  return 0;
}
```
and then you do:
```bash
cc main.c -o main
./main
```
It will print `WARNING: linker: /data/data/com.termux/files/home/main: unsupported flags DT_FLAGS_1=0x8000001`.
This happens because the executable has elf headers included. If you clean the executable and run it again
```bash
termux-elf-cleaner main
./main
```
Nothing will be printed.

To build V do this:
 ```bash
git clone https://github.com/vlang/v
cd v
make
./v symlink
```
And then you have it.

**Comment by larpon** [comment]
> Of course.
> Run those commands to setup
> 
> ```shell
> apt update && apt upgrade
> apt install git clang termux-elf-cleaner
> ```
> 
> If you create a, i.e. C file named main.c with the contents below
> 
> ```c
> int main() {
>   return 0;
> }
> ```
> 
> and then you do:
> 
> ```shell
> cc main.c -o main
> ./main
> ```
> 
> It will print `WARNING: linker: /data/data/com.termux/files/home/main: unsupported flags DT_FLAGS_1=0x8000001`.
> This happens because the executable has elf headers included. If you clean the executable and run it again
> 
> ```shell
> termux-elf-cleaner main
> ./main
> ```
> 
> Nothing will be printed.
> 
> To build V do this:
> 
> ```shell
> git clone https://github.com/vlang/v
> cd v
> make
> ./v symlink
> ```
> 
> And then you have it.

Thanks @fabriciopashaj

**Comment by larpon** [comment]
@medvednikov @spytheman @spaceface777 - I can probably fix this relatively easy. I'm a little unsure if this is even V's job in the first place? Cleaning up after specific compiler executables with thirdparty tools if detected? What do you think? What's the general stance on this?

**Comment by fabricio-p** [comment]
The main issue is that it's annoying, especially when using the repl. For each line that is evaluated, the warning is printed and the terminal becomes polluted. Maybe add an option like `-noelf`.
